### PR TITLE
Feature: Remove MCR floor

### PR DESCRIPTION
**Pull request by MilGard91**
## Context

#903 

## Changes proposed in this pull request

Remove MCR floor from the MCR contract

## Test plan

Please describe the tests cases that you ran to verify your changes. Add further instructions on 
how to run them if needed (i.e. migration / deployment scripts, env vars, etc).

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [ ] Performed a self-review of my own code
- [ ] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [ ] Didn't generate new warnings
- [ ] Didn't generate failures on existing tests
- [ ] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: contracts/modules/capital/MCR.sol [hunk 1]
```diff
@@ -122,26 +120,22 @@ contract MCR is IMCR, MasterAwareV2 {
   }
 
   /*
-  * @dev trigger an MCR update. Current virtual MCR value is synced to storage, mcrFloor is potentially updated
+  * @dev trigger an MCR update. Current virtual MCR value is synced to storage
```
Comment (roxdanila) [inline contracts/modules/capital/MCR.sol:123]: I couldn't comment above, as there are no changes, but can we please rename `getAllSumAssurance()` function to `getTotalCoverAmountInEth()` or similar?

File: test/unit/MCR/updateMCR.js [hunk 2]
```diff
@@ -323,23 +166,4 @@ describe('updateMCR', function () {
     const expectedDesiredMCR = totalSumAssured.mul(10000).div(gearingFactor);
     expect(desiredMCR).to.be.equal(expectedDesiredMCR);
   });
-
-  it('decreases desiredMCR towards mcrFloor when poolValueInEth = 0 and totalSumAssured = 0', async function () {
```
Comment (roxdanila) [inline test/unit/MCR/updateMCR.js:327]: what is the desiredMCR under these circumstances now, that we don't have a floor? do we end up having issues like division by 0? do they have any impact?

File: test/unit/MCR/updateMCR.js [hunk 3]
```diff
@@ -44,33 +41,6 @@ describe('updateMCR', function () {
     expect(lastUpdateTime).to.be.equal(previousLastUpdateTime);
   });
 
-  it('keeps values the same if MCR = MCR floor and mcrWithGear is too low', async function () {
```
Comment (roxdanila) [inline test/unit/MCR/updateMCR.js:47]: any new unit tests we should add to make sure removing the floor doesn't have weird consequences? they could be added in a later PR too, just worth thinking about it tho

File: test/unit/MCR/updateMCR.js (hunk 2 above)
Comment (shark0der) [reply test/unit/MCR/updateMCR.js:327]: The new desiredMCR becomes directly the gearedMCR

File: test/unit/MCR/updateMCR.js (hunk 3 above)
Comment (shark0der) [reply test/unit/MCR/updateMCR.js:47]: Previously it was driven by the maximum of the two, now it's going to be driven by mcr gear alone. Can't see any immediate issues. Please open an issue if this still seems uncertain.

**Review by shark0der — APPROVED** [review]

**Pull request merged**